src/mpi/comm/comm_split_type.c

```diff
@@ -346,21 +205,11 @@ int MPIR_Comm_split_type_node_topo(MPIR_Comm * user_comm_ptr, int split_type, in
         goto use_node_comm;
 
     /* if hw topology is not initialized, skip topology-aware comm split */
-    if (!MPIR_hw_topo_is_initialized())
+    if (!MPIR_hwtopo_is_initialized())
         goto use_node_comm;
 
     if (flag) {
-        if (!strncmp(hintval, "pci:", strlen("pci:")))
-            mpi_errno = node_split_pci_device(comm_ptr, key, hintval, newcomm_ptr);
-        else if (!strncmp(hintval, "ib", strlen("ib")) ||
-                 !strncmp(hintval, "en", strlen("en")) ||
-                 !strncmp(hintval, "eth", strlen("eth")) || !strncmp(hintval, "hfi", strlen("hfi")))
-            mpi_errno = node_split_network_device(comm_ptr, key, hintval, newcomm_ptr);
-        else if (!strncmp(hintval, "gpu", strlen("gpu")))
-            mpi_errno = node_split_gpu_device(comm_ptr, key, hintval, newcomm_ptr);
-        else
-            mpi_errno = node_split_processor(comm_ptr, key, hintval, newcomm_ptr);
-
+        mpi_errno = node_split(comm_ptr, key, hintval, newcomm_ptr);
```

Comment:
Can we use `hint_str` instead of `hintval`? I first thought it was `h-int-val` LOL ;) Then I had a hard time to note that here `val` is not a *numeric* value.

Reply:
This I can do :)